README.md

```diff
@@ -1,2 +1,437 @@
+    }
+
+    showCars() {
+      return this.props.cars.map((car) => (
+        <Car key={car.id} car={car}/>
+      ));
+    }
+
+    render() {
+      return (
+        <div>
+          {this.renderErrorMessage()}
+          {this.showCars()}
+        </div>
+      );
+    }
+  }
+  ```
+
+* No space before `/>`
+  ```js
+  // good
+  <Button text="Save" onChange={this.save}/>
+
+  // bad
+  <Button text="Save" onChange={this.save} />
+  ```
+
+* Closing `>` on its own line
+  ```js
+  // good
+  <Modal
+    title="Confirm Changes"
+    open={open}
+    onClose={this.closeModal}
+  >
+    <p>This action cannot be undone</p>
+    <Button text="Save" onChange={this.save}/>
+  </Modal>
+
+  // bad
+  <Modal
+    title="Confirm Changes"
+    open={open}
+    onClose={this.closeModal}>
+    <p>This action cannot be undone</p>
+    <Button text="Save" onChange={this.save}/>
+  </Modal>
+  ```
+* Self closing
+  ```js
+  // good
+  <Input
+    type="email"
+    label="Email"
+    placeholder="name@example.com"
+    onChange={this.setValue('email')}
+    onClick={this.selectText}/>
+
+  // bad
+  <Input
+    type="email"
+    label="Email"
+    placeholder="name@example.com"
+    onChange={this.setValue('email')}
+    onClick={this.selectText}
+  />
+  ```
+
+* Conditionally displayed elements should use the `&&` pattern. If the condition is not trivial it should be extracted.
+  ```js
+  <div>
+    {error && <div className="error">{error}</div>}
+  </div>
+  ```
+
+* Inline Map functions should use implicit return and parenthesis instead of explicit return and braces where possible.
+  ```js
+  // good
+  <ul>
+    {cars.map((car) => (
+      <li>{car.color}</li>
+    ))}
+  </ul>
+
+  // bad
+  <ul>
+    {cars.map((car) => {
+      return <li>{car.color}</li>;
+    })}
+  </ul>
+  ```
+
+## Whitespace
+
+* Object whitespace
+  ```js
+  // good
+  const car = {color: 'red', model: 'Saturn'};
+
+  // bad
+  const car = { color: 'red', model: 'Saturn' };
+  // bad
+  const car = {color:'red',model:'Saturn'};
+  ```
+
+* Import whitespace
+  ```js
+  // good
+  import { Card, Button } from 'semantic-ui-react';
+
+  // bad
+  import {Card, Button} from 'semantic-ui-react';
+  ```
+
+* Destructuring
+  ```js
+  // good
+  selectItem = ({target:{value:selectedItem=0}}) => this.setState({selectedItem})
+
+  // bad
+  selectItem = ({target: {value: selectedItem = 0}}) => this.setState({selectedItem})
+  // bad
+  selectItem = (e) => this.setState({selectedItem: e.target.value || 0})
+  ```
+
+* Collapse closing symbols
+  ```js
+  // good
+  cars.filter((car) => {
+    return car.color == 'red';
+  });
+
+  // less good
+  cars.filter(
+    (car) => {
+      return car.color == 'red';
+    }
+  );
+  ```
+
+## Methods
+
+* React component methods should be in [lifecycle order](https://reactjs.org/docs/react-component.html#the-component-lifecycle) with the exception of `render()` which should always be the last method.
```

Comment:
Good point. 
The ordering is actually really helpful because you can understand the entire lifecycle by reading top to bottom. Also, `ComponentWillUpdate` and `ComponentDidUpdate` are easily confused and having them in the right order helps to keep it straight